Sources/HTTP/Request/Request.swift

```diff
@@ -0,0 +1,66 @@
+import Core
+import URI
+
+public final class Request: Message {
+    public var method: Method
+    public var uri: URI
+    public var version: Version
```

Comment:
having all of these as publicly settable variables seems odd, should some of these be protected? ie: body, uri, version, shouldn't really be changing, method shouldn't be changing outside of internal, if at all.